eogrow/pipelines/testing.py

```diff
@@ -48,3 +57,82 @@ def run_procedure(self) -> Tuple[List, List]:
         LOGGER.info("The first EOPatch has a name %s", eopatches[0])
 
         return [], []
+
+
+class RasterFeatureSchema(BaseSchema):
+    feature: Feature = Field(description="A feature to be processed.")
+    shape: Tuple[int, ...] = Field(description="A shape of a feature")
+    dtype: str = Field(description="The output dtype of the feature")
+    min_value: int = Field(0, description="All values in the feature will be greater or equal to this value.")
+    max_value: int = Field(1, description="All values in the feature will be smaller to this value.")
+
+
+class TimestampFeatureSchema(BaseSchema):
+    time_period: TimePeriod = Field(description="Time period from where timestamps will be generated.")
+    _validate_time_period = field_validator("time_period", parse_time_period, pre=True)
+
+    timestamp_num: int = Field(description="Number of timestamps from the interval")
+
+
+class DummyDataPipeline(Pipeline):
+    """Pipeline for generating dummy data."""
+
+    class Schema(Pipeline.Schema):
+        output_folder_key: str = Field(description="The storage manager key pointing to the pipeline output folder.")
+        seed: Optional[int] = Field(description="A randomness seed.")
+
+        raster_features: List[RasterFeatureSchema]
+        timestamp_feature: Optional[TimestampFeatureSchema]
+
+    config: Schema
+
+    def build_workflow(self) -> EOWorkflow:
+        start_node = EONode(CreateEOPatchTask())
+
+        if self.config.timestamp_feature:
+            task = DummyTimestampFeatureTask(
+                time_interval=self.config.timestamp_feature.time_period,
+                timestamp_num=self.config.timestamp_feature.timestamp_num,
+            )
+            start_node = EONode(task, inputs=[start_node])
+
+        add_feature_nodes = []
+        for index, feature_config in enumerate(self.config.raster_features):
+            task = DummyRasterFeatureTask(
+                feature_config.feature,
+                shape=feature_config.shape,
+                dtype=np.dtype(feature_config.dtype),
+                min_value=feature_config.min_value,
+                max_value=feature_config.max_value,
+            )
+            node = EONode(task, inputs=[start_node], name=f"{DummyRasterFeatureTask.__name__}_{index}")
+            add_feature_nodes.append(node)
+
+        join_node = EONode(MergeEOPatchesTask(), inputs=add_feature_nodes)
```

Comment:
what if we want timestamps but no raster features? too much of an edge case?

Reply:
Does `DummyRasterFeatureTask` need the timestamp feature? Otherwise we could have them not depend on it and just join all of them with the merge task?

Reply:
Good catch - creating EOPatches with no raster features is a completely valid edge case which I missed. I fixed the code now and added an extra test for running a pipeline that generates only timestamps and bboxes.

It turns out that use `MergeEOPatchesTask` if you have time-dependent features but no timestamps. Therefore, timestamps have to be generated before raster features.